Use actual directive def'n instead of mock.

kegluneq opened the pull request:
Per earlier request & TODO in angular2/test/transform/common.dart

mhevery commented:
Please format your SHAs per https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format

mhevery commented on modules/angular2/test/transform/ctor_with_default_value_files/bar.dart:
```diff
@@ -2,8 +2,8 @@ library bar;
+class MyComponent {
   final dynamic c;
-  Component([this.c = 'sandwich']);
+  MyComponent([this.c = 'sandwich']);
```
An actual component will never have variable length parameters, as it is not compatible with DI.

mhevery commented on modules/angular2/test/transform/ctor_with_default_value_files/expected/index.bootstrap.dart:
```diff
@@ -5,10 +5,10 @@ import 'index.dart' as i2;
-    ..registerType(i0.Component, {
-      "factory": (dynamic c) => new i0.Component(c),
+    ..registerType(i0.MyComponent, {
+      "factory": (dynamic c) => new i0.MyComponent(c),
```
Could we give `c` property some type information and than show that the generated stub as the same type?

mhevery commented on modules/angular2/test/transform/list_of_types_files/bar.dart:
```diff
@@ -3,8 +3,8 @@ library bar;
 
-@Directive(context: const [MyContext])
-class Component {
+@Component(lightDomServices: const [MyContext])
```
`lightDomServices` are going away, use `componentServices` instead.

mhevery commented on modules/angular2/test/transform/two_deps_files/bar.dart:
```diff
@@ -3,11 +3,11 @@ library bar;
 
-@Directive(context: const MyContext(contextString))
-class Component2 {
+@Component(selector: const MyContext(contextString))
```
`selector` will always be a string. Not sure what `MyContext` represents here. This should be a analyzer error but we have to switch this types: https://github.com/angular/angular/blob/master/modules/angular2/src/core/annotations/annotations.js#L6

Commit:
refactor(dart/transform tests): Use actual directive def'n instead of mock.

In the transformer unit tests, we previously used a mock directive annotation. This update
substitutes the actual Angular2 directive annotations.

kegluneq commented:
I believe I've addressed all comments - PTAL.
